actionview/lib/action_view/helpers/tag_helper.rb

```diff
@@ -26,7 +26,100 @@ module TagHelper
+            else
+              content_tag_string(name, content || '', options, escape_attributes)
+            end
+          end
+
+          def respond_to_missing?(method_name, include_private = false)
+            true
+          end
+
+          def method_missing(called, *args, &block)
+            tag_string(called, *args, &block)
+          end
+      end
+
+      # Returns an HTML tag. Supports two syntax variants: legacy and modern.
+      #
+      # === Modern syntax
+      # Modern syntax follows one of two formats:
+      #   tag.<name>(options)
+      #   tag.<name>(content, options)
+      # Returns an HTML tag. Content has to be a string. If content is passed
+      # than tag is surrounding the content. Otherwise tag will be empty. You
+      # can also use a block to pass the content inside ERB templates. Result
+      # is by default HTML5 compliant. Include +escape_attributes+: +false+ 
+      # in options to disable attribute value escaping. The tag will be
+      # generated with related closing tag unless tag represents a
+      # void[https://www.w3.org/TR/html5/syntax.html#void-elements] element.
+      #
+      # ==== Options
+      # Like with traditional syntax the options hash can be used with
+      # attributes with no value (like disabled and readonly), which you can
+      # give a value of true in the options hash. You can use symbols or
+      # strings for the attribute names.
+      #    
+      # ==== Examples
+      #   tag.span
+      #   # => <span></span>
+      #
+      #   tag.span(class: "bookmark")
+      #   # => <span class=\"bookmark\"></span>
+      #
+      #   tag.input type: 'text', disabled: true
+      #   # => <input type="text" disabled="disabled">
+      #
+      #   tag.input type: 'text', class: ["strong", "highlight"]
+      #   # => <input class="strong highlight" type="text">
+      #
+      #   tag.img src: "open & shut.png"
+      #   # => <img src="open &amp; shut.png">
+      #
+      #   tag.img(src: "open & shut.png", escape_attributes: false)
+      #   # => <img src="open & shut.png">
+      #
+      #   tag.div(data: {name: 'Stephen', city_state: %w(Chicago IL)})
+      #   # => <div data-name="Stephen" data-city-state="[&quot;Chicago&quot;,&quot;IL&quot;]"></div>
+      #
+      #   tag.p "Hello world!"
+      #   # => <p>Hello world!</p>
+      #
+      #   tag.div tag.p("Hello world!"), class: "strong"
+      #   # => <div class="strong"><p>Hello world!</p></div>
+      #
+      #   tag.div "Hello world!", class: ["strong", "highlight"]
+      #   # => <div class="strong highlight">Hello world!</div>
+      #
+      #   tag.select options, multiple: true
+      #   # => <select multiple="multiple">...options...</select>
+      #
+      #   <%= tag.div class: "strong" do %>
+      #     Hello world!
+      #   <% end %>
+      #   # => <div class="strong">Hello world!</div>
+      #
+      #   <%= tag.div class: "strong" do |t| %>
+      #     <% tag.p("Hello world!") %>
```

Comment:
It needs to be in <%=  %>. `tag.div` returns string, but if you have a block as an argument it needs to be in ERB template, because it uses `CaptureHelper::capture` which requires `output_buffer` in the current context. Semantics is the same as for content_tag.